OpenRA.Game/Unforgiving.cs

```diff
@@ -0,0 +1,127 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+#nullable enable
+using System;
+using System.Reflection;
+
+namespace OpenRA
+{
+	/// <summary>
+	/// Provides null-unforgiving equivalents of methods,
```

Comment:
Given the `!` is called the null-forgiving operator, I choose "unforgiving" as a name for these methods that throw rather than returning null. But open to ideas on naming here.

In general this is here so if you've got an annoying method that returns nulls you don't care about, sticking an "Unforgiving" at the front/end will give you a non-null version.

Reply:
Fwiw, I saw this pattern with `ElseThrow` instead of `Unforgiving` in Java. (And although I hate Java, I think ElseThrow is a bit more explicit.)

Reply:
In rust terms `Some` perhaps, but that also doesn't make it clearer. If not many functions will be unforgiving you could consider returning an `Optional`/`Mono`. 

With your explenation `Unforgiven` may make the most sense. But needs getting used to.